src/coreclr/jit/ee_il_dll.hpp

```diff
@@ -44,6 +44,12 @@ void Compiler::eeGetFieldInfo(CORINFO_RESOLVED_TOKEN* pResolvedToken,
     info.compCompHnd->getFieldInfo(pResolvedToken, info.compMethodHnd, accessFlags, pResult);
 }
 
+FORCEINLINE
+uint32_t Compiler::eeGetThreadLocalFieldInfo(CORINFO_FIELD_HANDLE field)
+{
+    return info.compCompHnd->getThreadLocalFieldInfo(field);
+}
+
```

Comment:
What happens if you use win-arm64 altjit on win-x64? Does this (and elsewhere?) need `if (info.compMatchedVM)` handling?

Reply:
I don't think so. `getThreadLocalFieldInfo()` just returns the offset correctly. Here is the generated code for arm64 using altjit.

```asm
G_M23357_IG02:
            mov     w20, wzr
            mov     w21, wzr
            cmp     w19, #0
            bls     G_M23357_IG05
            ldr     x0, [xpr, #0x58]
            ldr     x0, [x0, #0x30]
            ldr     w1, [x0, #0x48]
            cmp     w1, #2
            blt     G_M23357_IG07
            ldr     x0, [x0, #0x50]
            ldr     x0, [x0, #0x10]
            cbz     x0, G_M23357_IG07
```